### PR TITLE
initial community

### PR DESCRIPTION
**Pull request by tobowers**
community: init!

### CODE REVIEW

File: community/cognitiveSteps/conversationNotes.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { createCognitiveStep, WorkingMemory, ChatMessageRoleEnum, indentNicely } from "@opensouls/engine";
+
+/**
```
Comment (kafischer) [inline community/cognitiveSteps/conversationNotes.ts:3]: These felt a little odd being bigger in the hierarchy than the basics

File: community/subprocesses/summarizeAndCompress.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    return workingMemory.slice(0, 1).withMemory({
+      role: ChatMessageRoleEnum.Assistant,
+      content: indentNicely`
+        ## Conversation so far
+        ${updatedNotes}
+      `,
+      metadata: {
+        conversationSummary: true
+      }
+    }).concat(workingMemory.slice(-2))
```
Comment (kafischer) [inline community/subprocesses/summarizeAndCompress.ts:43]: Is this amount of compression something you've been using? Seems more than what I've been copying from the souls templates?

File: community/subprocesses/summarizeAndCompress.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+
+import { ChatMessageRoleEnum, MentalProcess, indentNicely, useActions, useProcessMemory } from "@opensouls/engine";
+import { internalMonologue } from "../cognitiveSteps/basics/internalMonologue.js";
```
Comment (kafischer) [inline community/subprocesses/summarizeAndCompress.ts:3]: I wonder if we should just copy the steps in here too by convention? Will ppl realize to copy them too?

File: community/package.json [hunk 4]
```diff
@@ -15,9 +16,11 @@
     "prepublishOnly": "npm run build"
```
Comment (kafischer) [inline community/package.json:16]: We can kill this?

Comment (kafischer) [reply community/package.json:16]: Ie no publishing of this package?

**Review by kafischer — CHANGES_REQUESTED** [review]
Add license file?

Also do we want one more directly and a type for "CognitiveFunctions" which are functions that take in a working memory and return a working memory? useRag is like that right?

File: community/cognitiveSteps/conversationNotes.ts (hunk 1 above)
Comment (tobowers) [reply community/cognitiveSteps/conversationNotes.ts]: moved into a directory

File: community/subprocesses/summarizeAndCompress.ts (hunk 2 above)
Comment (tobowers) [reply community/subprocesses/summarizeAndCompress.ts]: i removed this file, we actually need to look at this a bit to make it more generic and not stomp on rag (or duplicate itself)

File: community/subprocesses/summarizeAndCompress.ts (hunk 3 above)
Comment (tobowers) [reply community/subprocesses/summarizeAndCompress.ts]: removed file, but yeah this is something we need to think about when there's a graph of dependencies.

**Comment by tobowers** [comment]
addressed these, some of them by removing stuff :)

**Comment by kafischer** [comment]
booooooo lol

**Review by kafischer — APPROVED** [review]

**Pull request merged**